src/main/php/lang/ast/syntax/PHP.class.php

```diff
@@ -1476,8 +1476,35 @@ private function annotations($parse, $context) {
     return $annotations;
   }
 
+  private function modifier($parse) {
+    $modifier= $parse->token->value;
+    $parse->forward();
+
+    if ('(' === $parse->token->value) {
+      $token= $parse->token;
+      $parse->expecting('(', 'modifiers');
+
+      // Disambiguate function types from `private(set)`
+      if ('function' === $parse->token->value) {
```

Comment:
The PHP compiler uses a different approach, parsing only `modifier("set")`, see https://github.com/php/php-src/pull/15063#discussion_r1686958867